src/util/samplebuffer.h

```diff
@@ -34,79 +35,94 @@ namespace mixxx {
 //
 class SampleBuffer final {
   public:
-    SampleBuffer()
-        : m_data(nullptr),
-          m_size(0) {
+    constexpr SampleBuffer()
+            : m_data(nullptr),
+              m_size(0) {
+    }
+    explicit SampleBuffer(SINT size)
+            : m_data((size > 0) ? SampleUtil::alloc(size) : nullptr),
+              m_size((m_data != nullptr) ? size : 0) {
     }
-    explicit SampleBuffer(SINT size);
     SampleBuffer(SampleBuffer&) = delete;
-    SampleBuffer(SampleBuffer&& that)
-        : m_data(that.m_data),
-          m_size(that.m_size) {
-        that.m_data = nullptr;
-        that.m_size = 0;
+    SampleBuffer(SampleBuffer&& that) noexcept
+            : m_data(std::exchange(that.m_data, nullptr)),
+              m_size(std::exchange(that.m_size, 0)) {
+    }
+    ~SampleBuffer() {
```

Comment:
TIL destructors are `noexcept` by default since C++11, so even if we don't add it here, the destructor would `std::terminate` be default if `SampleUtil::free` would panic.